### PR TITLE
Fix Coverity

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
This PR fixes:

<details> <summary>Coverity issues</summary>
<pre>
** CID 502561:         (Y2K38_SAFETY)
/src/ml/ml_kmeans.cc: 27           in ml_kmeans_train(ml_kmeans_t *, const std::vector<dlib::matrix<double, (long)6, (long)1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>, std::allocator<dlib::matrix<double, (long)6, (long)1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>>> &, unsigned int, long, long)()
/src/ml/ml_kmeans.cc: 28           in ml_kmeans_train(ml_kmeans_t *, const std::vector<dlib::matrix<double, (long)6, (long)1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>, std::allocator<dlib::matrix<double, (long)6, (long)1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>>> &, unsigned int, long, long)()

_____________________________________________________________________________________________
*** CID 502561:           (Y2K38_SAFETY)
/src/ml/ml_kmeans.cc: 27             in ml_kmeans_train(ml_kmeans_t *, const std::vector<dlib::matrix<double, (long)6, (long)1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>, std::allocator<dlib::matrix<double, (long)6, (long)1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>>> &, unsigned int, long, long)()
21         kmeans->max_dist = std::numeric_limits<calculated_number_t>::min();
22     }
23     
24     void
25     ml_kmeans_train(ml_kmeans_t *kmeans, const std::vector<DSample> &preprocessed_features, unsigned max_iters, time_t after, time_t before)
26     {
>>>     CID 502561:           (Y2K38_SAFETY)
>>>     A "time_t" value is stored in an integer with too few bits to accommodate it.  The expression "after" is cast to "uint32_t".
27         kmeans->after = (uint32_t) after;
28         kmeans->before = (uint32_t) before;
29     
30         kmeans->min_dist = std::numeric_limits<calculated_number_t>::max();
31         kmeans->max_dist  = std::numeric_limits<calculated_number_t>::min();
32     
/src/ml/ml_kmeans.cc: 28             in ml_kmeans_train(ml_kmeans_t *, const std::vector<dlib::matrix<double, (long)6, (long)1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>, std::allocator<dlib::matrix<double, (long)6, (long)1, dlib::memory_manager_stateless_kernel_1<char>, dlib::row_major_layout>>> &, unsigned int, long, long)()
22     }
23     
24     void
25     ml_kmeans_train(ml_kmeans_t *kmeans, const std::vector<DSample> &preprocessed_features, unsigned max_iters, time_t after, time_t before)
26     {
27         kmeans->after = (uint32_t) after;
>>>     CID 502561:           (Y2K38_SAFETY)
>>>     A "time_t" value is stored in an integer with too few bits to accommodate it.  The expression "before" is cast to "uint32_t".
28         kmeans->before = (uint32_t) before;
29     
30         kmeans->min_dist = std::numeric_limits<calculated_number_t>::max();
31         kmeans->max_dist  = std::numeric_limits<calculated_number_t>::min();
32     
33         kmeans->cluster_centers.clear();
```
</pre>
</details>

##### Test Plan

- Check CI;
- Compile on your environment and verify ML is running;

<!--
Provide enough detail so that your reviewer can understand which test cases you
have covered, and recreate them if necessary. If our CI covers sufficient tests, then state which tests cover the change.
-->

##### Additional Information

<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Fixes Coverity Y2K38 warnings by using time_t for k-means time bounds end-to-end and hardening timestamp handling. Prevents truncation and rejects invalid timestamps in JSON and SQLite.

- **Bug Fixes**
  - Replace `uint32_t` with `time_t` for `after`/`before` in `ml_kmeans_t` and `ml_kmeans_inlined_t`; remove casts in `ml_kmeans_train`.
  - Deserialize `after`/`before` with `json_object_get_int64`, enforcing non-negative values that fit in `time_t`.
  - Persist/load `after`/`before` via SQLite as 64-bit (`sqlite3_bind_int64`, `sqlite3_column_int64`); add bounds checks on load and skip out-of-range rows; update pruning to use `time_t`.
  - Add unit tests to round-trip > INT32_MAX timestamps on 64-bit `time_t` and to reject negative timestamps.

<sup>Written for commit 6f3b74fd09104c1beba86ed95e980035518ff0d6. Summary will update on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

### CODE REVIEW

File: src/ml/ml_kmeans.h [hunk 1]
```diff
@@ -13,8 +13,8 @@ struct ml_kmeans_t {
     std::vector<DSample> cluster_centers;
     calculated_number_t min_dist;
     calculated_number_t max_dist;
-    uint32_t after;
-    uint32_t before;
+    time_t after;
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply src/ml/ml_kmeans.h:18]: Address with https://github.com/netdata/netdata/pull/22094/commits/31e7f32b1bcdc4708616e2fa97a8b36d1715bdd3.

File: src/ml/ml_kmeans.h [hunk 2]
```diff
@@ -13,8 +13,8 @@ struct ml_kmeans_t {
...
+    time_t before;
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply src/ml/ml_kmeans.h:19]: Fixed with https://github.com/netdata/netdata/pull/22094/commits/6f3b74fd09104c1beba86ed95e980035518ff0d6

File: src/ml/ml_kmeans.cc [hunk 3]
```diff
@@ -186,7 +186,7 @@ bool ml_kmeans_deserialize(ml_kmeans_inlined_t *inlined_km, struct json_object *
         netdata_log_error("Failed to deserialize kmeans: failed to parse int for 'after'");
         return false;
     }
-    inlined_km->after = json_object_get_int(value);
+    inlined_km->after = (time_t) json_object_get_int64(value);
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply src/ml/ml_kmeans.cc:196]: Already fixed.

File: src/ml/ml_kmeans.cc [hunk 4]
```diff
@@ -196,7 +196,7 @@ bool ml_kmeans_deserialize(ml_kmeans_inlined_t *inlined_km, struct json_object *
         netdata_log_error("Failed to deserialize kmeans: failed to parse int for 'before'");
         return false;
     }
-    inlined_km->before = json_object_get_int(value);
+    inlined_km->before = (time_t) json_object_get_int64(value);
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply src/ml/ml_kmeans.cc:212]: Everything you suggested was already fixed.
